ctlcmd/boardcmd.py

```diff
@@ -0,0 +1,70 @@
+"""
+  boardcmd.py
+
+  Commands that are used for modifying the board and its conditions. 
+  This includes loading and saving the coordinates calibration results.
+"""
+
+import ctlcmd.cmdbase as cmdbase
+import cmod.fmt as fmt
+import argparse
+
+class save_board(cmdbase.controlcmd):
+  """
+  @brief Saving current board session.
+  """
+  def __init__(self, cmd):
+    cmdbase.controlcmd.__init__(self, cmd)
+
+  def add_args(self):
+    self.parser.add_argument('--filename',
+                             '-f',
+                             type=argparse.FileType(mode='r'),
+                             help="""
+                             Overwrite the board configuration json filename to save current board session.""",
+                             required=False)
+
+  def run(self, args):
+    """
+    For the sake of clarity, device settings is split into each of their
+    functions. Notice that all function should have exception guards so the
+    subsequent settings can still be set if settings for one particular device
+    is bad or not available.
+    """
+    try:
+      if args.filename:
+        self.board.save_board(args.filename)
+      else:
+        self.board.save_board()
+    except RuntimeError as err:
+      self.printerr(str(err))
+      self.printwarn('Board saving has failed, skipping...')
+
+class load_board(cmdbase.controlcmd):
+  """
+  @brief Loading board from given config filename.
+  """
+  def __init__(self, cmd):
+    cmdbase.controlcmd.__init__(self, cmd)
+
+  def add_args(self):
+    self.parser.add_argument('--filename',
+                             '-f',
+                             type=argparse.FileType(mode='r'),
```

Comment:
This will result in 
```bash
> load_board -f ./cfg/latest_board.json
[ERROR  ][load_board] expected str, bytes or os.PathLike object, not TextIOWrapper
``` 

So change you will need to either have board.load_board accept file descriptors or have this argument only accept strings

Reply:
good catch here. Thank you.